exporter/prometheusremotewriteexporter/exporter.go

```diff
@@ -229,7 +228,7 @@ func (prwe *prwExporter) handleExport(ctx context.Context, tsMap map[string]*pro
 	}
 
 	// Calls the helper function to convert and batch the TsMap to the desired format
-	requests, err := batchTimeSeries(tsMap, prwe.maxBatchSizeBytes, m, &prwe.batchTimeSeriesState)
+	requests, err := batchTimeSeries(tsMap, prwe.maxBatchSizeBytes, m, prwe.batchTimeSeriesState)
```

Comment:
> What happens if there are multiple batches processed concurrently that have significantly different sizes?

Good point; they would still share the same state, and their results would be conflicting. I think I need to go back to the drawing board and think a bit more about how to solve this

Reply:
Actually, that was a lot simpler :)

Reply:
I'm rereading the code, and my understanding is that concurrent requests with very distinct batch sizes would constantly fight for the size of the subsequent request.

Can we even do something useful with the batchStateSize if we allow multiple workers? It sounds like this optimization only works for a single worker scenario 🤔

Reply:
> I'm rereading the code, and my understanding is that concurrent requests with very distinct batch sizes would constantly fight for the size of the subsequent request.

That's how I understand it, as well.  I also think it likely that each of the three sizes tracked by this state would be decorrelated, though I'm not sure that's any more problematic.
 
> Can we even do something useful with the batchStateSize if we allow multiple workers? It sounds like this optimization only works for a single worker scenario 🤔

I'm not sure this optimization is safe with multiple workers.  Would it make more sense to use a `sync.Pool` of backing stores that can eventually grow to the needed size and get periodically reaped to avoid one-off large batches causing leaks?  Similar to what is done in #35184?

Reply:
Using the sync.Pool sounds like worth exploring! We could also remove the state altogether and see how bad the benchmarks will look.

I'm trying things out and running benchmarks, I'll open new PRs once I have something to show :)